### ISSUE TITLE
componentWIllUnmount gets called before navigation onBlur callback

### ISSUE THREAD

**Issue by Meninblack007** [software-mansion/react-native-screens#1695]
### Description

With `react-native-screens` disabled, we used to get `blur` event on componentWillUnmount. Attaching screenshot for lifecycle.

[
<img width="284" alt="screens disabled" src="https://user-images.githubusercontent.com/6505547/214373120-020b1edc-11cf-4212-9cf7-17beaecdee8c.png">
](url)

With `react-native-screens` enabled, we don't get `blur` event on componentWillUnmount as we tend to remove listener on unmount and unmount happens before blur callback is received by the container. Attaching screenshot for lifecycle.

<img width="277" alt="screen enabled" src="https://user-images.githubusercontent.com/6505547/214373612-1772afb1-248b-4b7d-ba3d-580f1fd4db2c.png">

### Steps to reproduce

Sharing snack links with repro (these snack links uses latest version of screens and react native but this issue exists on older versions as well)
JS navigator: https://snack.expo.dev/@sanyam.jain/js-navigator
Native navigator: https://snack.expo.dev/@sanyam.jain/native-navigator

### Snack or a link to a repository

https://snack.expo.dev/@sanyam.jain/native-navigator

### Screens version

3.11.1

### React Native version

0.63.4

### Platforms

Android, iOS

### JavaScript runtime

None

### Workflow

None

### Architecture

None

### Build type

None

### Device

None

### Acknowledgements

Yes

**Comment by Meninblack007** [comment]
@WoLewicki @kkafar help will be appreciated.

**Comment by kkafar** [comment]
@Meninblack007 Have you tried more recent version of `react-native-screens`. Does the issue persist there?

**Comment by Meninblack007** [comment]
@kkafar Yes. Locally I have tried with `react-native-screens` 3.19.0 and the issue still exists.

**Comment by Meninblack007** [comment]
Hi @kkafar 

Were you able to recreate with the snack provided?

**Comment by kkafar** [comment]
Yes I did, however on native side things seems to be working properly & I need to consult with react-navigation team regarding some details of the JS side of implementation. 

I'll get back to you if there is some progress

**Comment by Meninblack007** [comment]
Sure. Waiting for an update. Thanks!

**Comment by kkafar** [comment]
@Meninblack007

It looks like we won't fix this on `react-native-screens` side. 

AFAIK when you use `navigation.pop()` / `navigation.goBack()` etc. the component (view) is removed from JS view tree, and only after this the information (that view has been removed) is passed down to the native side -> native view hierarchy is modified, transition animation is played and native events are sent back to JS side, but there is no receiver for them (the component was already unmounted). It seems that there is no workaround that can be done in native `react-native-screens` code. You can try to take this issue to `react-navigation`, maybe they can emit some artificial event from JS. 

I'm closing the issue due to aforementioned reasons, but please feel free to tag me and continue the discussion.

**Comment by Meninblack007** [comment]
Thanks @kkafar. I will raise this issue over there.
